pkg/controllers/managementuser/secret/secret.go

```diff
@@ -68,6 +100,135 @@ func Register(ctx context.Context, cluster *config.UserContext) {
+	}
+
+	return nil
+}
+
+func (dsc *DirectSyncController) syncable(obj *corev1.Secret) bool {
+	// no sync annotations, we don't care about this secret
+	if obj.Annotations[syncAnnotation] == "" && obj.Annotations[syncPreBootstrapAnnotation] == "" {
+		return false
+	}
+
+	// if secret is authorized to be synchronized to the cluster
+	if !slices.Contains(strings.Split(obj.Annotations[capr.AuthorizedObjectAnnotation], ","), dsc.clusterName) {
+		return false
+	}
+
+	// if the secret is not in a namespace that we are allowed to sync to
+	if !slices.Contains(approvedPreBootstrapTargetNamespaces, obj.Annotations[syncNamespaceAnnotation]) {
+		return false
+	}
+
+	return true
+}
+
+func (dsc *DirectSyncController) bootstrapSyncable(obj *corev1.Secret) bool {
+	// only difference between sync and bootstrapSync is requiring the boostrap sync annotation to be set to "true"
+	return dsc.syncable(obj) && obj.Annotations[syncPreBootstrapAnnotation] == "true"
+}
+
+func (dsc *DirectSyncController) sync(key string, obj *corev1.Secret) (runtime.Object, error) {
+	if obj == nil {
+		return nil, nil
+	}
+
+	if !dsc.syncable(obj) {
+		return obj, nil
+	}
+
+	name := obj.Annotations[syncNameAnnotation]
+	if name == "" {
+		name = obj.Name
+	}
+	ns := obj.Annotations[syncNamespaceAnnotation]
+	if ns == "" {
+		ns = obj.Namespace
+	}
+
+	logrus.Debugf("[direct-sync][secret] synchronizing %v/%v to %v/%v for cluster %v", obj.Namespace, obj.Name, ns, name, dsc.clusterName)
+
+	op := "update"
+	// fetching every time due to the fact that we don't have a downstream cache in the upstream cluster anymore
+	targetSecret, err := dsc.downstreamSecrets.GetNamespaced(ns, name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logrus.Debugf("[direct-sync][secret] secret %v/%v not found for for cluster %v, creating", ns, name, dsc.clusterName)
+			op = "create"
+		} else {
+			return nil, fmt.Errorf("failed to get downstream secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
+		}
+	}
+
+	if targetSecret != nil && reflect.DeepEqual(targetSecret.Data, obj.Data) {
+		logrus.Debugf("[direct-sync][secret] skipping downstream update - contents are the same")
+		return obj, nil
+	}
+
+	if op == "create" {
+		logrus.Debugf("[direct-sync][secret] creating secret %v/%v in cluster %v", ns, name, dsc.clusterName)
+
+		targetSecret, err = dsc.downstreamSecrets.Create(
+			&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: ns},
+				Data:       obj.Data,
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
+		}
+	} else {
+		logrus.Debugf("[direct-sync][secret] updating secret %v/%v in cluster %v", ns, name, dsc.clusterName)
+
+		targetSecret.Data = obj.Data
+		_, err = dsc.downstreamSecrets.Update(targetSecret)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
+		}
+	}
```

Comment:
```suggestion
	var targetSecret *corev1.Secret
	if targetSecret, err = dsc.downstreamSecrets.GetNamespaced(ns, name, metav1.GetOptions{}); err != nil && !errors.IsNotFound(err) {
		return nil, fmt.Errorf("failed to get downstream secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
  }
  
  if targetSecret == nil {
  	logrus.Debugf("[direct-sync][secret] creating secret %v/%v in cluster %v", ns, name, dsc.clusterName)

		_, err = dsc.downstreamSecrets.Create(
			&corev1.Secret{
				ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: ns},
				Data:       obj.Data,
			},
		)
		if err != nil {
			return nil, fmt.Errorf("failed to create secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
  } else if !reflect.DeepEqual(targetSecret.Data, obj.Data) {
    logrus.Debugf("[direct-sync][secret] updating secret %v/%v in cluster %v", ns, name, dsc.clusterName)

		targetSecret.Data = obj.Data
		_, err = dsc.downstreamSecrets.Update(targetSecret)
		if err != nil {
			return nil, fmt.Errorf("failed to update secret %v/%v in cluster %v: %v", ns, name, dsc.clusterName, err)
  } else {
		logrus.Debugf("[direct-sync][secret] skipping downstream update - contents are the same")
		return obj, nil
  }
```
String values are enums are probably not what we want for this, I feel like the above is much simpler. The comment removals are not on purpose.